worlds/sc2/__init__.py

```diff
@@ -680,6 +680,7 @@ def fill_pool_with_kerrigan_levels(world: SC2World, item_pool: List[Item]):
     if (world.options.kerrigan_presence.value not in kerrigan_unit_available
         or total_levels == 0
         or not kerrigan_missions
+        or SC2Race.ZERG not in get_enabled_races(world)
```

Comment:
Recommend removing this check here; I've generally tried very hard to remove post-mission selection generation on these "exclude X missions" options. Possibilities here:
* On line 678 right above, exclude The Infinite Cycle from `kerrigan_missions`
* Set `grant_story_levels` to true in your test that's running afoul of this check
* Set `grant_story_levels` to true by default; it may be the better UX experience overall as hots players will adjust it to taste while people playing other campaigns will ignore it, in which case story levels is the expected behaviour

Reply:
After running locally and debugging, I found the issue: on line 679, `kerrigan_build_missions` includes _all_ build missions, not just kerrigan ones. Should be:
```python
kerrigan_build_missions = [mission for mission in kerrigan_missions if MissionFlag.NoBuild not in mission.flags]
```

Reply:
Thanks for catching and fixing that! I'll remove line 683 and merge your changes.